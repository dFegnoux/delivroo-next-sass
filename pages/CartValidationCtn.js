import React, { Component } from "react";
import DeliveryForm from "../components/forms/DeliveryForm";
import Cart from "../components/menu/Cart";
import PageHelmet from "../components/PageHelmet";

class Menus extends Component {
  state = {
    cart: {},
    showCardForm: false
  };

  render() {
    const { cart } = this.state;

    return (
      <div className="menuLayout">
        <PageHelmet title="Validate your order" />
        <div className="menuContentWrapper">
          <DeliveryForm />
        </div>
        <Cart menus={cart} disableButtons />
      </div>
    );
  }

  componentDidMount() {
    const retrievedCart = window.localStorage.getItem("cart");
    if (retrievedCart) {
      this.setState(() => ({
        cart: JSON.parse(retrievedCart)
      }));
    }
  }
}

export default Menus;
