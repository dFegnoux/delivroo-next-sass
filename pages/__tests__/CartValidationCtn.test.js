import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import CartValidationCtn from "../CartValidationCtn";

describe("CartValidationCtn lifecylce", () => {
  afterEach(() => {
    localStorage.removeItem("cart");
  });
  it("Should retrieve cart from localStorage", () => {
    const cartToRetrieve = {
      storageCartItemId: { title: "storage item", quantity: 1 }
    };
    localStorage.setItem("cart", JSON.stringify(cartToRetrieve));
    const container = shallow(<CartValidationCtn />);
    expect(container.state("cart")).toEqual(cartToRetrieve);
  });
});
