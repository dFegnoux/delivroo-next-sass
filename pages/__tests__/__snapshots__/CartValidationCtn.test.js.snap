// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartValidationCtn snapshots should render correctly 1`] = `
<div
  className="menuLayout"
>
  <div
    className="menuContentWrapper"
  >
    <form
      className="paymentForm"
      onSubmit={[Function]}
    >
      <div
        className="input col-6"
      >
        <label>
          Etage et numéro d'appartement
        </label>
        <input
          name="flat"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="ex : Appartement n°15"
          type="text"
        />
      </div>
      <div
        className="input col-6"
      >
        <label>
          Digicode
        </label>
        <input
          name="code"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="ex : B123"
          type="text"
        />
      </div>
      <div
        className="input col-12"
      >
        <label>
          Adresse
        </label>
        <input
          name="street_address"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="ex : 100 Rue de Rivoli"
          type="text"
        />
      </div>
      <div
        className="input col-6"
      >
        <label>
          Code postal
        </label>
        <input
          name="postcode"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="ex : 75001"
          type="text"
        />
      </div>
      <div
        className="input col-6"
      >
        <label>
          Ville
        </label>
        <input
          name="city"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="ex : Paris"
          type="text"
        />
      </div>
      <div
        className="input col-12"
      >
        <label>
          Numéro de téléphone
        </label>
        <input
          name="phone"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="+33 9 77 55 03 30"
          type="text"
        />
      </div>
      <div
        className="input col-12"
      >
        <label>
          Instructions pour votre livreur ?
        </label>
        <input
          name="instructions"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="ex : C'est la porte noire près de l'animalerie. Merci d'appeler lorsque vous arrivez."
          type="text"
        />
      </div>
      <button
        className="btn [object Object] "
        disabled={true}
        onClick={null}
        type="submit"
      >
        Valider
      </button>
    </form>
  </div>
  <div
    className="cartWrapper"
  >
    <div
      className="cart roundedItem"
    >
      <div>
        Votre panier est vide
      </div>
    </div>
  </div>
</div>
`;
